src/thermobench.cpp

```diff
@@ -203,6 +204,35 @@ void write_arr_csv(FILE *fp, double *arr, int size)
     fprintf(fp, "\n");
 }
 
+void wait_cooldown(char* exec_dir){
+
+    FILE *fp = fopen(state.sensors[0].path, "r");
+    if (fp == NULL)
+        err(1, "Error while opening sensor file: %s\n", state.sensors[0].path);
+
+    char* cmd;
+    asprintf(&cmd,"%s/fan_ctrl.sh 1", exec_dir);
+    if(system(cmd) == -1)
+        err(1, "Error while executing shell command: %s\n", cmd);
+
+    while(1){
+        rewind(fp);
+        int temp = INT_MAX;
+        if(fscanf(fp, "%d", &temp) < 1) continue;
+        if(temp <= cooldown_temp) break;
+        printf("Cooling down to %d, CPU temperature: %d\n",cooldown_temp,temp);
+        fflush(stdout);
+        sleep(1);
+    }
+
+    cmd[strlen(cmd)-1]='0';
+    if(system(cmd) == -1)
+        err(1, "Error while executing shell command: %s\n", cmd);
+
```

Comment:
You forgot to `free()` cmd. Maybe, you can use C++ string which frees itself automatically and makes it easier to append the argument with `+` operator.